userspace/libsinsp/container.cpp

```diff
@@ -74,7 +74,8 @@ bool sinsp_container_manager::remove_inactive_containers()
 			return true;
 		});
 
-		for(auto it = m_containers.begin(); it != m_containers.end();)
```

Comment:
This change seems to be about enabling synchronized access to the set of containers and not getting resource limits from cgroups. Is that the intent?

Reply:
It's actually both, sorry about that. I'd split this in two but the parts are mostly independent so it hopefully won't affect the review too much